File: CS-IOS/Classes/Classes/Network/CSNetwork.m
//
// Created by Rene Dohan on 02/09/17.
// Copyright (c) 2017 renetik_software. All rights reserved.
//

#import "CSNetwork.h"
#import <SystemConfiguration/CaptiveNetwork.h>


@implementation CSNetwork

+ (NSString *)getSSID {
    CFArrayRef interFaceNames = CNCopySupportedInterfaces();
    CFDictionaryRef networkInfo = CNCopyCurrentNetworkInfo(CFArrayGetValueAtIndex(interFaceNames, 0));
    if (networkInfo) return (__bridge_transfer id) CFDictionaryGetValue(networkInfo, kCNNetworkInfoKeySSID);
    return @"";
}

@end

File: Renetik/Renetik/Classes/Core/framework/CSNetwork.swift
//
//  CSNetwork.swift
//  Renetik
//
//  Created by Rene Dohan on 2/21/19.
//

import SystemConfiguration.CaptiveNetwork

public class CSNetwork: NSObject {
    @objc public class var SSID: String? {
        if let interfaces = CNCopySupportedInterfaces() as NSArray? {
            for interface in interfaces {
                if let interfaceInfo =
                CNCopyCurrentNetworkInfo(interface as! CFString) as NSDictionary? {
                    return interfaceInfo[kCNNetworkInfoKeySSID as String] as? String
                }
            }
        }
        return nil
    }
}
